### PR TITLE
Refactor minitest to RSpec

### PR DESCRIPTION
**Pull request by cjsim89**

### CODE REVIEW

File: spec/customer_spec.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    expect(@joel.outstanding_balance).to eq(0)
+  end
+
+  it "can charge to increase oustanding balance" do
+    @joel.charge(15)
+    @joel.charge(7)
+
+    expect(@joel.outstanding_balance).to eq(22)
+  end
+
```
Comment (dcoleman21) [inline spec/customer_spec.rb:41]: Maybe delete this blank line (41)

File: spec/day_care_spec.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    molly = Pet.new({name: "Molly", type: :cat})
+    billy.adopt(molly)
+
+    @day_care.add_customer(joel)
+    @day_care.add_customer(billy)
+
+    lucy.feed
+
+    expect(@day_care.unfed_pets).to eq([samson, molly])
+
```
Comment (dcoleman21) [inline spec/day_care_spec.rb:50]: Get rid of this blank space (47). Do we want to put these `Pets` and `Customers` in the before(:each)?

File: spec/day_care_spec.rb [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    joel = Customer.new("Joel", 2)
+    joel.charge(10)
+    billy = Customer.new("Billy", 3)
+
+    @day_care.add_customer(joel)
+    @day_care.add_customer(billy)
+
+    expect(@day_care.customer_by_id(2)).to eq(joel)
+  end
+
```
Comment (dcoleman21) [inline spec/day_care_spec.rb:62]: Get rid of this blank space (60).

File: spec/groomer_spec.rb [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    @groomer.add_customer(billy)
+
+    expect(@groomer.customers).to eq([joel, billy])
+  end
+
+  it "can count the number of pets of a certain type" do
+    joel = Customer.new("Joel", 2)
+    samson = Pet.new({name: "Samson", type: :dog})
+    lucy = Pet.new({name: "Lucy", type: :cat})
+    joel.adopt(samson)
```
Comment (dcoleman21) [inline spec/groomer_spec.rb]: Again, do we want to put `Pets` and `Customers` into the before(:each)?

File: spec/groomer_spec.rb [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  it "can list customers with outstanding balances" do
+    joel = Customer.new("Joel", 2)
+    joel.charge(10)
+    billy = Customer.new("Billy", 3)
+    @groomer.add_customer(joel)
+    @groomer.add_customer(billy)
+
+    expect(@groomer.customers_with_oustanding_balances).to eq([joel])
+  end
+
```
Comment (dcoleman21) [inline spec/groomer_spec.rb]: Get rid of this blank line (55).

File: spec/pet_spec.rb [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+  it "is not fed by default" do
+    expect(@samson.fed?).to be(false)
+  end
+
+  it "can be fed" do
+    @samson.feed
+    expect(@samson.fed?).to be(true)
+  end
+
```
Comment (dcoleman21) [inline spec/pet_spec.rb:25]: Get rid of this extra space (25).

**Review by dcoleman21 — COMMENTED** [review]
Hey Chris, I left my thoughts on things. Feel free to use them or not. Thank you for taking the time to switch this over to Rspec.

File: spec/customer_spec.rb (hunk 1 above)
Comment (rtillies) [reply spec/customer_spec.rb:41]: Atom automatically inserts a blank line at the end of most (all?) files by default. The blank line at the end of a file is common and oftentimes recommended. This places the EOF marker on its own line and separates it from the newline character at the end of the last line of actual code.

File: spec/day_care_spec.rb (hunk 2 above)
Comment (rtillies) [reply spec/day_care_spec.rb:50]: The `DayCare` object needs to contain no customers at first. Can you create nested `before :each do` blocks; as in, can you have one `before` block for testing the empty array, then another `before` block for the `pet` and `customer` objects for the remaining tests?

**Review by rtillies — COMMENTED** [review]
Looks good overall. I added my comments to Dani's comments. Thanks for doing this!

File: spec/day_care_spec.rb (hunk 2 above)
Comment (dcoleman21) [reply spec/day_care_spec.rb:50]: If we put all of the objects in the `before(:each)` we would only need to add the `add_customer(joel)` in the tests that need it. Does that make sense?

File: spec/customer_spec.rb (hunk 1 above)
Comment (dcoleman21) [reply spec/customer_spec.rb:41]: I learned something new. I didn't know that atom did that. Cool.

File: spec/groomer_spec.rb (hunk 4 above)
Comment (cjsim89) [reply spec/groomer_spec.rb]: Whoops, forgot to do the same change in this file that I did in the day_care. Will fix.

File: spec/day_care_spec.rb (hunk 2 above)
Comment (cjsim89) [reply spec/day_care_spec.rb:50]: Just updated this - check out the latest changes.

Comment (dcoleman21) [reply spec/day_care_spec.rb:50]: Love the additional `before(:each)` that @rtillies suggested. Thanks for amending this @cjsim89

**Review by dcoleman21 — COMMENTED** [review]
Fantastic @cjsim89. Thank you for doing this.

**Pull request merged**